kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java

```diff
@@ -165,8 +172,32 @@ public List<KieContainerResource> buildDeployments(KieServerProperties kieServer
     public List<KieContainerResource> buildAutoScanDeployments(KieServerProperties kieServerProperties) throws IOException {
         ApplicationHome appHome = new ApplicationHome();
         final String folder = "BOOT-INF/classes/KIE-INF/lib/";
-        File root = appHome.getSource();
-        return discoverDeployments(folder, new FileInputStream(root));
+        return (explodedJarFolder == null) ? discoverDeployments(folder, new FileInputStream(appHome.getSource())) :  discoverDeployments(folder, new File(explodedJarFolder));
+    }
+
+    public List<KieContainerResource> discoverDeployments(String folder, File root) {
+        try {
+            Path base = Paths.get(root.toString(), folder);
+            if(!Files.isDirectory(base)) {
+                logger.error("Autoscan folder failed. {} does not exist or it is not a folder", base);
+                return Collections.emptyList();
+            }
+            List<KieContainerResource> files = new ArrayList<>();
+            List<Path> candidates = Files.walk(base).filter(Files::isRegularFile).collect(Collectors.toList());
```

Comment:
Use try-with-resources or close this "Stream" in a "finally" clause.

Reply:
done